src/Sylius/Bundle/CoreBundle/CatalogPromotion/Calculator/PercentageDiscountPriceCalculator.php

```diff
@@ -27,7 +27,7 @@ public function supports(CatalogPromotionActionInterface $action): bool
 
     public function calculate(ChannelPricingInterface $channelPricing, CatalogPromotionActionInterface $action): int
     {
-        $price = (int) ($channelPricing->getPrice() - ($channelPricing->getPrice() * $action->getConfiguration()['amount']));
+        $price = (int) round($channelPricing->getPrice() - ($channelPricing->getPrice() * $action->getConfiguration()['amount']));
```

Comment:
As it's a change in `PercentageDiscountPriceCalculator` service implementation, we would need to improve its spec as well 🖖